src/components/Pagination/Pagination.tsx

```diff
@@ -32,145 +32,106 @@ export type PaginationProps = {
-      pageNumber: 0,
-      createURL: this.props.createURL,
-    });
+    props.setCurrentPage(pageNumber);
   };
 
-  private lastPageLink = () => {
-    return this.pageLink({
-      ariaLabel: 'Last',
-      additionalClassName: this.props.cssClasses.lastPageItem,
-      isDisabled: this.props.isLastPage,
-      label: this.props.templates.last,
-      pageNumber: this.props.nbPages! - 1,
-      createURL: this.props.createURL,
-    });
-  };
-
-  private pages = () => {
-    return this.props.pages!.map((pageNumber) =>
-      this.pageLink({
-        ariaLabel: `${pageNumber + 1}`,
-        additionalClassName: this.props.cssClasses.pageItem,
-        isSelected: pageNumber === this.props.currentPage,
-        label: `${pageNumber + 1}`,
-        pageNumber,
-        createURL: this.props.createURL,
-      })
-    );
-  };
-
-  public render() {
-    return (
-      <div
-        className={cx(this.props.cssClasses.root, {
-          [this.props.cssClasses.noRefinementRoot]: this.props.nbPages! <= 1,
-        })}
-      >
-        <ul className={this.props.cssClasses.list}>
-          {this.props.showFirst && this.firstPageLink()}
-          {this.props.showPrevious && this.previousPageLink()}
-          {this.pages()}
-          {this.props.showNext && this.nextPageLink()}
-          {this.props.showLast && this.lastPageLink()}
-        </ul>
-      </div>
-    );
-  }
+  const pages = props.pages!.map((pageNumber) => (
+    <PaginationLink
+      key={pageNumber}
+      ariaLabel={`${pageNumber + 1}`}
+      additionalClassName={props.cssClasses.pageItem}
+      isSelected={pageNumber === props.currentPage}
+      label={`${pageNumber + 1}`}
+      pageNumber={pageNumber}
+      createURL={props.createURL}
+      cssClasses={props.cssClasses}
+      handleClick={handleClick}
+    />
+  ));
+
+  return (
+    <div
+      className={cx(props.cssClasses.root, {
+        [props.cssClasses.noRefinementRoot]: props.nbPages! <= 1,
+      })}
+    >
+      <ul className={props.cssClasses.list}>
+        {props.showFirst && (
+          <PaginationLink
+            key="first"
+            ariaLabel={'First'}
+            additionalClassName={props.cssClasses.firstPageItem}
+            isDisabled={props.isFirstPage}
+            label={props.templates.first}
+            pageNumber={0}
+            createURL={props.createURL}
+            cssClasses={props.cssClasses}
+            handleClick={handleClick}
+          />
+        )}
+
+        {props.showPrevious && (
+          <PaginationLink
+            key="previous"
+            ariaLabel={'Previous'}
+            additionalClassName={props.cssClasses.previousPageItem}
+            isDisabled={props.isFirstPage}
+            label={props.templates.previous}
+            pageNumber={props.currentPage - 1}
+            createURL={props.createURL}
+            cssClasses={props.cssClasses}
+            handleClick={handleClick}
+          />
+        )}
+
+        {pages}
```

Comment:
Not blocking, but since we're not using `pages` anywhere else, it could be inlined so the code is visually in the same order as the visual representation.

Reply:
sure, this was just closer to the previous implementation, I prefer it inline too